pkg/sanity/sanity.go

```diff
@@ -135,15 +158,23 @@ func (sc *SanityContext) setup() {
 	}
 
 	By("creating mount and staging directories")
-	err = createMountTargetLocation(sc.Config.TargetPath)
-	Expect(err).NotTo(HaveOccurred())
-	if len(sc.Config.StagingPath) > 0 {
-		err = createMountTargetLocation(sc.Config.StagingPath)
-		Expect(err).NotTo(HaveOccurred())
-	}
+
+	// If callback function for creating target dir is specified, use it.
+	targetPath, err := createMountTargetLocation(sc.Config.TargetPath, sc.Config.CreateTargetPathCmd, sc.Config.CreateTargetDir, sc.Config.CreatePathCmdTimeout)
+	Expect(err).NotTo(HaveOccurred(), "failed to create target directory %s", targetPath)
+	sc.targetPath = targetPath
+
+	// If callback function for creating staging dir is specified, use it.
+	stagingPath, err := createMountTargetLocation(sc.Config.StagingPath, sc.Config.CreateStagingPathCmd, sc.Config.CreateStagingDir, sc.Config.CreatePathCmdTimeout)
+	Expect(err).NotTo(HaveOccurred(), "failed to create staging directory %s", stagingPath)
+	sc.stagingPath = stagingPath
 }
 
 func (sc *SanityContext) teardown() {
+	// Delete the created paths if any.
+	os.RemoveAll(sc.targetPath)
+	os.RemoveAll(sc.stagingPath)
```

Comment:
I added this in teardown because without this, some tests were using the directories created by the previous tests and that was resulting in the directory exists check in `createMountTargetLocation()` to be true and hence, not returning a target path even if it was created was passing, because that code block wasn't being executed.
With this, the directories are deleted for every test and not returning target path from `createMountTargetLocation()` results in failure.

Reply:
While it's nicer if the test cleans up after itself, we then also need functions that correspond to `CreateStagingDir` and `CreateTargetDir`. The main use-case for these functions is for testing a driver that was deployed somewhere, and then those functions will not create the dirs locally.

Reply:
Right! I wasn't thinking in that way. Thanks, I'll add the corresponding cleanup options as well.